File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task82_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.Base64;

public class Main {
    public static void main(String[] args) throws NoSuchAlgorithmException {
        String password = "mypassword";
        String salt = "mysalt";
        String combined = password + salt;
        MessageDigest md = MessageDigest.getInstance("SHA-256");
        byte[] hashBytes = md.digest(combined.getBytes());
        String hash = Base64.getEncoder().encodeToString(hashBytes);
        System.out.println("Hash: " + hash);
    }
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task82_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <iomanip>
#include <openssl/sha.h>

std::string computeHash(const std::string& password, const std::string& salt) {
    std::string combined = password + salt;
    unsigned char hash[SHA256_DIGEST_LENGTH];
    SHA256((unsigned char*)combined.c_str(), combined.size(), hash);
    std::stringstream ss;
    for (int i = 0; i < SHA256_DIGEST_LENGTH; i++) {
        ss << std::hex << std::setw(2) << std::setfill('0') << (int)hash[i];
    }
    return ss.str();
}

int main() {
    std::string password = "mysecretpassword";
    std::string salt = "mysalt";
    std::cout << computeHash(password, salt) << std::endl;
    return 0;
}